packages/data-stores/src/plans/queries/use-plans.ts

```diff
@@ -27,6 +28,13 @@ function usePlans( {
 	coupon && params.append( 'coupon_code', coupon );
 	params.append( 'locale', locale );
 
+	if ( typeof window !== 'undefined' ) {
+		const currency = getQueryArg( window.location.search, 'currency' );
+		if ( typeof currency === 'string' ) {
+			params.append( 'currency', currency );
+		}
+	}
```

Comment:
I wonder if we should instead pass this through as a param and avoid the side-effect and dependency. For whatever reason, this hook may be used in a context where `currency` in the URL should not translate into a change here. WDYT?

Reply:
Thanks for taking a look @chriskmnds.

> What should happen when you are on /plans/[foo]?currency=INR and click to switch internal?

Just to clarify, what does 'click to switch internal' refer to?  The only other way I know of to change the currency is via store admin. My rough thinking is that whenever `?currency=INR` is present it would take priority.

> I wonder if we should instead pass this through as a param and avoid the side-effect and dependency

I had the same thought, and actually implemented it as a param to `usePlans` initially. 

However because my motivation for this change was as a sort of 'developer feature' to help with testing and PR reviews, I thought it might be better to avoid components having an awareness of this param 🤔.

It does raise a question in my mind though, about whether this feature will be used beyond just testing and PR reviews. If yes, then I think it makes sense to implement it as a `param` that can be used by any component.

Out of interest @niranjan-uma-shankar, for the `/pricing` page, did you intend for the `currency` param to be used by other features, or just to help with testing?  eg. someone clicks a button in an email that loads the pricing page with a particular currency.

Reply:
>  did you intend for currency param support on /pricing to be used by other features, or just to help with testing?

I only considered this from the angle of making it easier to test multiple currencies. I imagine in all other scenarios,  we would just go with the currency associated with the user in their user attribute.

Reply:
> Just to clarify, what does 'click to switch internal' refer to? The only other way I know of to change the currency is via store admin. My rough thinking is that whenever ?currency=INR is present in the browser URL it would take priority over the store admin setting.

Sorry, I meant to switch "interval" (or the "term"), not "internal". If you are view 1-year plans with the currency param, then click the dropdown to view 2-year plans (the currency param is not passed along).

All in all, considering that this is only meant for testing on the development side and that we can already do that through SA, and that this code will not follow through on redirects/route changes (like changing intervals from the dropdown), then I fail to see the value here. I'm not even sure if even someone who would want to test multiple currencies would prefer to update manually a URL (and again if they click on a route change) instead of the dropdown in SA. I would reconsider the nature of this change and whether something we want to maintain in production (more so on essential hooks that derive pricing).